File: data/problems/TwoSum/cpp/solution_expected.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> visited;
        
        for (int i = 0; i < nums.size(); ++i) {
            const int curElement = nums[i];
            const int delta = target - curElement;
            
            const auto it = visited.find(delta);
            if (it != visited.end()) {
                return {it->second, i};
            }
            
            visited.insert({curElement, i});
            continue;

        }
        return {};
    }
};

File: data/problems/TwoSum/python/solution_expected.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {}

        for index, value in enumerate(nums):
            delta = target - value
            if delta in visited:
                return [visited[delta], index]
            visited[value] = index

        return []
